.github/workflows/code-quality.yaml

```diff
@@ -0,0 +1,96 @@
+name: ✨ Code Quality Check
+on:
+  push:
+    branches:
+      - main
+      - dev
+  pull_request: {}
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
```

Comment:
Based on the on events in workflow, changes might be needed here.

Reply:
could you elaborate? what changes?

Reply:
The github.ref does not exist in every workflow run, such as in pull_request. 

So, how about the following configuration?

```
group: ${{ github.workflow }}-${{ github.ref || github.head_ref || github.run_id }}
cancel-in-progress: true
```

Reply:
> The github.ref does not exist in every workflow run, such as in pull_request.
> 
> So, how about the following configuration?
> 
> ```
> group: ${{ github.workflow }}-${{ github.ref || github.head_ref || github.run_id }}
> cancel-in-progress: true
> ```

For workflows triggered by pull_request, `github.ref` is the pull request merge branch.
`github.run_id` is unique for every workflow run and wouldn't serve the purpose of canceling previous workflows.
`github.head_ref` is only available in `pull_request` or `pull_request_target`, which is even more limited than `github.ref`.

Reply:
Got it, you are right. It looks like using `github.ref` should be sufficient. 
I was mistaken here.